qubesbuilder/executors/__init__.py

```diff
@@ -114,52 +115,76 @@ def replace_placeholders(self, s: str):
         return s
 
     @staticmethod
-    async def _read_stream(stream, cb, max_length=10000):
+    async def _read_stream(stream, callback, max_length=10000) -> bytes:
         remaining_line = b""
+        buffer = b""
```

Comment:
This buffering is a bit unfortunate, especially if not all callers need it. The output can be quite big (for example kernel build is several megabytes). Maybe make it optional?